languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml

```diff
@@ -38791,6 +38791,71 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
 
 
     <category id='COMPOUNDING' name="Palavras Compostas" type="misspelling">
+        <rule id="SEM_ABRIGO" name="Composição de sem-abrigo">
+            <pattern>
+                <token postag_regexp="yes" postag="[ADP].+M[SP].+|Z.+|SENT_START"></token>
```

Comment:
I'm a bit weary of including adjectives here, I feel like some FPs will happen. Hopefully few enough to deal with with antipatterns. For example: `Eu com casa e o bonito sem abrigo`

Also with `DP*`:

`O meu em casa e o seu sem abrigo`

Maybe adjectives and possessives should be dealt with separately.

Reply:
Hm, you're not wrong, I'll remove adjectives from here and see what happens.